_posts/2025-03-31-zscore-hybrid-search.md

```diff
@@ -0,0 +1,170 @@
+
+## What is Z-Score Normalization?
+
+Z-score normalization, also known as standardization, is a method of scaling data using mean and standard deviation. The formula for calculating the Z-score is:
+Z = (X - μ) / σ
+Where:
+
+* X is the original value
+* μ is the mean of the population
+* σ is the standard deviation of the population
+
+## When to use Z Score?
+
+Considering your index’s structure can help you decide which one to choose since each has advantages of its own. If your documents are more similar to one another and the top-k results of a typical query return documents that are very similar to one another and clustered together within the index, as seen in the graph below, Min-Max may be a better option.
+
+![Image for min max distribution](/assets/media/blog-images/2025-03-31-zscore-hybrid-search/blogpost1.jpg){: .img-fluid}
+
+However, Z-Score is more suited if the results are more evenly distributed and have some characteristics of a normal distribution, as shown in the example below.
+
+![Image for zscore distribution](/assets/media/blog-images/2025-03-31-zscore-hybrid-search/blogpost2.jpg){: .img-fluid}
+
+The basic flow to use between min max and z score looks like below:
+
+![Image for flow](/assets/media/blog-images/2025-03-31-zscore-hybrid-search/blogpost3.png){: .img-fluid}
+
+### How to use Z Score?
+
+To use z_score, create a search pipeline and specify `z_score` as the `technique`:
+
+```
+PUT /_search/pipeline/z_score-pipeline
+{
+    "description": "Zscore processor for hybrid search",
+    "phase_results_processors": [
+        {
+            "normalization-processor": {
+                "normalization": {
+                    "technique": "z_score"
+                },
+                "combination": {
+                    "technique": "arithmetic_mean"
+                }
+            }
+        }
+    ]
+}
+```
+
+Next, create a hybrid query and apply the pipeline to it
+
+```
+
+
+POST my_index/_search?search_pipeline=z_score-pipeline
+{
+   "query": {
+     "hybrid": [
+         {}, // First Query
+         {}, // Second Query
+         ... // Other Queries
+     ] 
+   }
+}
+```
+
+
+
+## Benchmarking Z Score performance
+
+Benchmark experiments were conducted using an OpenSearch cluster consisting of a single r6g.8xlarge instance as the coordinator node, along with three r6g.8xlarge instances as data nodes. To assess Z Score’s performance comprehensively, we measured three key metrics across four distinct datasets. For information about the datasets used, see [Datasets](https://public.ukp.informatik.tu-darmstadt.de/thakur/BEIR/datasets/).
+
+Search relevance was quantified using the industry-standard Normalized Discounted Cumulative Gain at rank 10 (NDCG@10). We also tracked system performance using search latency measurements and monitored CPU utilization to analyze resource consumption during the experiments. This setup provided a strong foundation for evaluating both search quality and operational efficiency.
```

Comment:
Is there supposed to be a table for the CPU utilization?

Reply:
Since both are score based techniques, cpu utilization won't have major difference

Reply:
Can we either remove the monitored CPU utilization part here then or mention that in the conclusions?

Reply:
Done